glide-core/redis-rs/redis/tests/test_cluster_async.rs

```diff
@@ -121,6 +121,165 @@ mod cluster_async {
         .unwrap();
     }
 
+    #[cfg(feature = "valkey-gte-8")]
+    #[tokio::test]
+    async fn test_routing_by_slot_to_replica_with_az_affinity_strategy_to_half_replicas() {
+        let replica_num: u16 = 4;
+        let primaries_num: u16 = 3;
+        let replicas_num_in_client_az = replica_num / 2;
+        let cluster =
+            TestClusterContext::new((replica_num * primaries_num) + primaries_num, replica_num);
+        let az: String = "us-east-1a".to_string();
+
+        let mut connection = cluster.async_connection(None).await;
+        let cluster_addresses: Vec<_> = cluster
+            .cluster
+            .servers
+            .iter()
+            .map(|server| server.connection_info())
+            .collect();
+
+        let mut cmd = redis::cmd("CONFIG");
+        cmd.arg(&["SET", "availability-zone", &az.clone()]);
+
+        for _ in 0..replicas_num_in_client_az {
+            connection
+                .route_command(
+                    &cmd,
+                    RoutingInfo::SingleNode(SingleNodeRoutingInfo::SpecificNode(Route::new(
+                        12182, // foo key is mapping to 12182 slot
+                        SlotAddr::ReplicaRequired,
+                    ))),
+                )
+                .await
+                .unwrap();
+        }
+
+        let mut client = ClusterClient::builder(cluster_addresses.clone())
+            .read_from(redis::cluster_slotmap::ReadFromReplicaStrategy::AZAffinity(
+                az.clone(),
+            ))
+            .build()
+            .unwrap()
+            .get_async_connection(None)
+            .await
+            .unwrap();
+
+        // Each replica in the client az will return the value of foo n times
+        let n = 4;
+        for _ in 0..n * replicas_num_in_client_az {
+            let mut cmd = redis::cmd("GET");
+            cmd.arg("foo");
+            let _res: RedisResult<Value> = cmd.query_async(&mut client).await;
+        }
+
+        let mut cmd = redis::cmd("INFO");
+        cmd.arg("ALL");
+        let info = connection
+            .route_command(
+                &cmd,
+                RoutingInfo::MultiNode((MultipleNodeRoutingInfo::AllNodes, None)),
+            )
+            .await
+            .unwrap();
+
+        let info_result = redis::from_owned_redis_value::<HashMap<String, String>>(info).unwrap();
+        let get_cmdstat = format!("cmdstat_get:calls={}", n);
+        let client_az = format!("availability_zone:{}", az);
+
+        let matching_entries_count = info_result
+            .values()
+            .filter(|value| value.contains(&get_cmdstat) && value.contains(&client_az))
+            .count();
+
+        assert_eq!(
```

Comment:
dont you want to check that others instances did not get GET commands?

Reply:
added